addons/hr_payroll_account/hr_payroll_account.py

```diff
@@ -114,16 +113,42 @@ def process_sheet(self, cr, uid, ids, context=None):
                 amt = slip.credit_note and -line.total or line.total
                 if float_is_zero(amt, precision_digits=precision):
                     continue
-                partner_id = line.salary_rule_id.register_id.partner_id and line.salary_rule_id.register_id.partner_id.id or default_partner_id
-                debit_account_id = line.salary_rule_id.account_debit.id
-                credit_account_id = line.salary_rule_id.account_credit.id
+                partner_id = False
+                # choose the account to create the debit move
+                if (line.salary_rule_id.debit_employee_account == 'receivable'
+                        and slip.employee_id.address_home_id
+                        and slip.employee_id.address_home_id.property_account_receivable):
+                    debit_account_id = slip.employee_id.address_home_id.property_account_receivable.id
+                    partner_id = slip.employee_id.address_home_id.id
+                elif (line.salary_rule_id.debit_employee_account == 'payable'
+                        and slip.employee_id.address_home_id
+                        and slip.employee_id.address_home_id.property_account_payable):
+                    debit_account_id = slip.employee_id.address_home_id.property_account_payable.id
+                    partner_id = slip.employee_id.address_home_id.id
+                elif line.salary_rule_id.debit_employee_account == 'other':
+                    debit_account_id = line.salary_rule_id.account_debit.id
+                else:
+                    debit_account_id = False
+                # choose the account to create the credit move
+                if (line.salary_rule_id.credit_employee_account == 'payable'
+                        and slip.employee_id.address_home_id
+                        and slip.employee_id.address_home_id.property_account_payable):
+                    credit_account_id = slip.employee_id.address_home_id.property_account_payable.id
+                    partner_id = slip.employee_id.address_home_id.id
+                elif (line.salary_rule_id.credit_employee_account == 'receivable'
+                        and slip.employee_id.address_home_id
+                        and slip.employee_id.address_home_id.property_account_receivable):
+                    credit_account_id = slip.employee_id.address_home_id.property_account_receivable.id
+                elif line.salary_rule_id.credit_employee_account == 'other':
```

Comment:
line.salary_rule_id.account_credit.id does not crash, it just returns False.
Anyway I'm adding a test to be sure it doesn"t crash and an attrs in the view to set the account as required if 'other' is selected